qds_sdk/util.py

```diff
@@ -142,3 +142,31 @@ def underscore(word):
     """
     return re.sub(r'\B((?<=[a-z])[A-Z]|[A-Z](?=[a-z]))',
                   r'_\1', word).lower()
+
+def _make_minimal(dictionary):
+    """
+    This function removes all the keys whose value is either None or an empty
+    dictionary.
+    """
+    new_dict = {}
+    for key, value in dictionary.items():
+        if value is not None:
+            if isinstance(value, dict):
+                new_value = _make_minimal(value)
+                if new_value:
+                    new_dict[key] = new_value
+            else:
+                new_dict[key] = value
+    return new_dict
+
+def _read_file(file_path):
+    file_content = None
+    if file_path is not None:
+        try:
+            with open(file_path) as f:
+                file_content = f.read()
+        except IOError as e:
+            sys.stderr.write("Unable to read %s: %s\n" % (file_path, str(e)))
+            sys.exit(1)
```

Comment:
I think you should not exit here. Instead you should throw IOError here and let its caller handle the exception (if at all).